File: LowLevelNetworking (Sockets)/Java/SimpleClient.java
package Random;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.net.Socket;

public class SimpleClient {

    private static String HOSTNAME = "localhost";
    private static int PORT = 80;

    public static void startClient() throws IOException {

        // using ARM (Automated resource Manager, so that I don't have to manually close the socket and BufferedReader/Writer)
        try (Socket socket = new Socket(HOSTNAME, PORT);
             BufferedReader br = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream()));) {

            // important to send a newline (\r\n) with each message
            bw.write("Hello Server!\r\n");
            bw.flush();
            System.out.println("Client Send!");

            String received = br.readLine();
            System.out.printf("Client Received: %s", received);
        }
    }

    public static void main(String... args) throws IOException {
        SimpleClient.startClient();
    }
}

File: LowLevelNetworking (Sockets)/C#/Simple_Client.cs
using System;
using System.Text;
using System.Net;
using System.Net.Sockets;
using System.IO;

class Simple_Client {

    private const string HOSTNAME = "localhost";
    private const int PORT = 80;

    public Simple_Client() { }

    public void startClient() {
        IPHostEntry ipHostInfo = Dns.GetHostEntry(HOSTNAME);
        IPAddress ipAddress = ipHostInfo.AddressList[0];
        IPEndPoint remoteEP = new IPEndPoint(ipAddress, PORT);

        Socket client = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Unspecified);

        try {
            client.Connect(remoteEP);

            using (NetworkStream stream = new NetworkStream(client))
            using (StreamReader sr = new StreamReader(stream))
            using (StreamWriter sw = new StreamWriter(stream)) {

                sw.WriteLine("Hello Server!\n");
                sw.Flush();
                Console.WriteLine("Client Send!");

                string received = sr.ReadLine();
                Console.WriteLine("Client Received: {0}", received);

                //client.Shutdown(SocketShutdown.Both);
                //client.Close();
            }
        }
        catch (Exception e) { Console.WriteLine(e.ToString()); }
    }

}
